src/lib/core/fiber.c

```diff
@@ -2394,5 +2396,6 @@ fiber_shutdown(double timeout)
 		}
 	}
 	cord()->shutdown_fiber = NULL;
+	trigger_run(&fiber_on_shutdown, NULL);
```

Comment:
Let's add this trigger in the previous commit (the one that adds `fiber_wait_dead`). BTW it looks like all subsystems using this trigger to exactly the same work. Maybe, instead of generic triggers, better provide a more narrow API that would take a fiber to stop on shutdown?

Reply:
Discussed f2f and decided to add a flag like `FIBER_IS_MANAGED_SHUTDOWN` that can be set for system fibers. If set then just shutdown this fiber on `fiber_shutdown` just as non system fibers.